fuzz/clamav_scanfile_fuzzer.cpp

```diff
@@ -117,21 +101,28 @@ ClamAVState kClamAVState;
 
 extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size)
 {
-    FILE* fuzzfile = NULL;
+    FILE* fuzzfile          = NULL;
+    char tmp_file_name[200] = {0};
+
+    __pid_t pid = getpid();
 
-    fuzzfile = fopen(kClamAVState.tmp_file_name, "w");
+    snprintf(tmp_file_name, sizeof(tmp_file_name), "tmp.scanfile.%d", pid);
+
+    fuzzfile = fopen(tmp_file_name, "w");
     fwrite(data, size, 1, fuzzfile);
     fclose(fuzzfile);
 
     const char* virus_name = nullptr;
     unsigned long scanned  = 0;
 
     cl_scanfile(
-        kClamAVState.tmp_file_name,
+        tmp_file_name,
         &virus_name,
         &scanned,
         kClamAVState.engine,
         &kClamAVState.scanopts);
 
+    unlink(tmp_file_name);
```

Comment:
Always good to check the return value from `unlink()`, just in case you need to fail because something horrible went wrong during the test (like a path being moved out from under you).

Reply:
I don't know that there's anything we can do if it fails. @mmmds any recommendations here?